recipes/quill/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+from conans import ConanFile, CMake, tools
+from conans.model.version import Version
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class QuillConan(ConanFile):
+    name = "quill"
+    description = "C++14 Asynchronous Low Latency Logging Library"
+    homepage = "https://github.com/odygrd/quill/"
+    topics = ("conan", "quill", "logging", "log")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fmt": [True, False]}
+    default_options = {"fmt": False}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        supported_archs = ["x86", "x86_64"]
+
+        if not any(arch in str(self.settings.arch) for arch in supported_archs):
+            raise ConanInvalidConfiguration("{} is not supported by {}".format(self.settings.arch, self.name))
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "14")
+
+        minimum_version = {
+            "clang": "5",
+            "gcc": "5",
+            "Visual Studio": "15.0",
+        }.get(str(self.settings.compiler))
+
+        if not minimum_version:
+            self.output.warn(
+                "Unknown compiler {} {}. Assuming compiler supports C++14".format(self.settings.compiler,
+                                                                                  self.settings.compiler.version))
+
+        else:
+            version = Version(self.settings.compiler.version.value)
+            if version < minimum_version:
+                raise ConanInvalidConfiguration(
+                    "The compiler {} {} does not support C++14".format(self.settings.compiler,
+                                                                       self.settings.compiler.version))
+
+    def requirements(self):
+        if self.options.fmt:
+            self.requires("fmt/6.1.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["QUILL_FMT_EXTERNAL"] = self.options.fmt
+
+        self._cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
```

Comment:
```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```

Reply:
This should fix your fmt error.

BTW use quill:fmt=True as option, not fmt=True. The package is part of the option name.

Reply:
Thanks a lot @uilianries : )
I will try again tonight.

Also in the CI build there is a different error. I am trying to fix that also before pushing to remote as this will trigger the CI again.

In the CI error in this log https://c3i.jfrog.io/c3i/misc/logs/pr/1416/5/quill/1.2.1/90ee443cae5dd5c1b4861766ac14dc6fae231a92/

[here](https://c3i.jfrog.io/c3i/misc/logs/pr/1416/5/quill/1.2.1/90ee443cae5dd5c1b4861766ac14dc6fae231a92/create_stdout.txt)

I can see that the library was build and installed, and then I see that the test_package is failed to build. It looks like a linker error.

However in my test_package  CMakeLists.txt i just link against `&{CONAN_LIBS}`

`
----Running------
> cd '/tmp/c3ipr/pr_1416_5_0_0/test_package/build/04a89eb3fd1ca69c0373770cfc8222ea2544fde6' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_IN_LOCAL_CACHE="OFF" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="5" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Wno-dev '/tmp/c3ipr/pr_1416_5_0_0/test_package'
-----------------
`

If you have any clue on this error and what it is wrong, it would be very much appreciated, thanks!